teachertool/src/components/HomeScreen.tsx

```diff
@@ -112,11 +173,76 @@ const GetStarted: React.FC = () => {
     );
 };
 
+interface DataCarouselProps {
+    title: string;
+    cardsUrl: string;
+}
+
+const CardCarousel: React.FC<DataCarouselProps> = ({ title, cardsUrl }) => {
+    const [cardSet, setCardSet] = useState<CarouselCardSet | undefined>();
+    const [fetchStatus, setFetchStatus] = useState<RequestStatus | undefined>();
+
+    useJsonDocRequest(cardsUrl, setFetchStatus, setCardSet);
+
+    return (
+        <>
+            <div className={css.carouselRow}>
+                <div className={css.rowTitle}>
+                    <h2>{title}</h2>
+                </div>
+                {(fetchStatus === "loading" || fetchStatus === "error") && (
+                    <Carousel>
+                        <LoadingCard />
+                        <LoadingCard delay={true} />
+                    </Carousel>
+                )}
+                {fetchStatus === "success" && (
+                    <Carousel>
+                        {cardSet?.cards.map((card, index) => {
+                            switch (card.cardType) {
```

Comment:
This feels messy to me. I recently learned about type guards and I'm wondering if this situation would be a good use case for it. https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates

Reply:
Moving the typecast out of the render does make the code more readable!